### PR TITLE
proxystore_test: Add unit test to SessionAffinity

### PR DESCRIPTION
**Pull request by dougsland**
Creates a unit test to SessionAffinity

GitHub-Issue: https://github.com/kubernetes-sigs/kpng/issues/156
Signed-off-by: Douglas Schilling Landgraf <dlandgra@redhat.com>
Signed-off-by: Jay Vyas (jayunit100) <jvyas@vmware.com>

### CODE REVIEW

**Comment by dougsland** [comment]
# go test -v triggered issue#156

**Comment by dougsland** [comment]
```
# go test -v
=== RUN   TestSessionAffinitySetClientIP
--- FAIL: TestSessionAffinitySetClientIP (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x7bad3b]

goroutine 20 [running]:
testing.tRunner.func1.2({0x838ec0, 0xc6ca00})
	/usr/local/go/src/testing/testing.go:1209 +0x24e
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1212 +0x218
panic({0x838ec0, 0xc6ca00})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/gogo/protobuf/proto.makeOneOfMarshaler.func1({0xc000152200}, 0x1)
	/root/go/pkg/mod/github.com/gogo/protobuf@v1.3.2/proto/table_marshal.go:2598 +0x7b
github.com/gogo/protobuf/proto.(*marshalInfo).size(0xc000152200, {0xc000088ab0})
	/root/go/pkg/mod/github.com/gogo/protobuf@v1.3.2/proto/table_marshal.go:200 +0x187
github.com/gogo/protobuf/proto.makeMessageMarshaler.func1({0xc000152180}, 0x1)
	/root/go/pkg/mod/github.com/gogo/protobuf@v1.3.2/proto/table_marshal.go:2430 +0x39
github.com/gogo/protobuf/proto.(*marshalInfo).size(0xc000152180, {0xc00016c690})
	/root/go/pkg/mod/github.com/gogo/protobuf@v1.3.2/proto/table_marshal.go:200 +0x187
github.com/gogo/protobuf/proto.(*InternalMessageInfo).Size(0x203000, {0x956358, 0xc00016c690})
	/root/go/pkg/mod/github.com/gogo/protobuf@v1.3.2/proto/table_marshal.go:131 +0x4a
github.com/gogo/protobuf/proto.(*Buffer).Marshal(0xc0001caa80, {0x956358, 0xc00016c690})
	/root/go/pkg/mod/github.com/gogo/protobuf@v1.3.2/proto/table_marshal.go:2990 +0x385
sigs.k8s.io/kpng/server/pkg/proxystore.(*Store).hashOf(0xc00016c5f0, {0x956358, 0xc00016c690})
	/root/kpng/server/pkg/proxystore/proxystore.go:69 +0x33
sigs.k8s.io/kpng/server/pkg/proxystore.(*Tx).SetService(0xc000138618, 0xc000172180, {0xc000119b70, 0x1, 0x1})
	/root/kpng/server/pkg/proxystore/proxystore.go:254 +0xad
sigs.k8s.io/kpng/server/pkg/proxystore.TestSessionAffinitySetClientIP.func1(0x501353)
	/root/kpng/server/pkg/proxystore/proxystore_test.go:87 +0x339
sigs.k8s.io/kpng/server/pkg/proxystore.(*Store).Update(0xc00016c5f0, 0x8df360)
	/root/kpng/server/pkg/proxystore/proxystore.go:89 +0xc6
sigs.k8s.io/kpng/server/pkg/proxystore.TestSessionAffinitySetClientIP(0x0)
	/root/kpng/server/pkg/proxystore/proxystore_test.go:86 +0x205
testing.tRunner(0xc0001d4680, 0x8df368)
	/usr/local/go/src/testing/testing.go:1259 +0x102
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1306 +0x35a
exit status 2
FAIL	sigs.k8s.io/kpng/server/pkg/proxystore	0.009s
```

**Comment by dougsland** [comment]
The unit test is failing as expected. Now we need to dig and solve it in a second patch (IMO).

**Comment by jayunit100** [comment]
Beautiful!!!!!!!!

File: server/pkg/proxystore/proxystore_test.go [hunk 1]
```diff
@@ -77,3 +78,45 @@ func Example() {
 	// - IPs:{V4:"10.0.0.2"} [  ]
 
 }
+
+// TestSessionAffinitySetClientIP creates scenario to validate SessionAffinity
+// Ref: https://github.com/kubernetes-sigs/kpng/issues/156
+func TestSessionAffinitySetClientIP(t *testing.T) {
+	store := New()
+	store.Update(func(tx *Tx) {
```
Comment (jayunit100) [inline server/pkg/proxystore/proxystore_test.go:86]: Whats this `Tx` thing?

**Review by jayunit100 — CHANGES_REQUESTED** [review]

File: server/pkg/proxystore/proxystore_test.go (hunk 1 above)
Comment (jayunit100) [reply server/pkg/proxystore/proxystore_test.go:86]: nvm i see

File: server/pkg/proxystore/proxystore_test.go [hunk 2]
```diff
@@ -77,3 +78,45 @@ func Example() {
 	// - IPs:{V4:"10.0.0.2"} [  ]
 
 }
+
+// TestSessionAffinitySetClientIP creates scenario to validate SessionAffinity
+// Ref: https://github.com/kubernetes-sigs/kpng/issues/156
+func TestSessionAffinitySetClientIP(t *testing.T) {
```
Comment (jayunit100) [inline server/pkg/proxystore/proxystore_test.go:84]: `func UnCommaneMeAmimTestSessionAffinitySetClientIP(t *testing.T) {`

Comment (jayunit100) [reply server/pkg/proxystore/proxystore_test.go:84]: then we'll merge #171 right after :), and Uncomment this test.

Comment (dougsland) [reply server/pkg/proxystore/proxystore_test.go]: sure thing

**Comment by jayunit100** [comment]
We'll uncomment the test when Amim's fix merges

**Pull request merged**